auth/credentials/clientcredentials.go

```diff
@@ -28,7 +29,7 @@ type tokenSource struct {
 
 // RevokeToken revokes the Access Token while also removing it from the Access Token Cache.
 // When the Access Token is expired or missing, revoke will return without any action.
-func (c *Config) RevokeToken(ctx context.Context, t *oauth2.Token) error {
+func (c *Config) RevokeToken(ctx context.Context, t *auth.Token) error {
```

Comment:
We removed logic that accepted token string. 
To pass token we will require client to store expiration or parse token into auth.Token.

Reply:
yes, I'm not a fan of the previous approach parsing the string to guess the expiry date, specially when that parsing wasn't doing any sort of signature validation of the token so they could easily be tampered strings, I strongly belive that we shouldn't be dealing with strings but a Token object, this is similar to what the official oauth lib does

Reply:
That means that customer applications will need to be dealing with building Token (from storage etc.)
Tampered strings/Token do not matter as all oauth backend does provide security features in this case. 
Client side uses expiration to avoid sending expired tokens only.

Reply:
the token type is (de)serializable, it'll be safer to operate within the struct than string parsing of the token and risk a tampered one, even if the back-end would still catch it

Reply:
SGTM